src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,89 @@
+    private static final int DECIMAL_FORMAT = 100;
+    private static final double DEFAULT_CIRCLE_RADIUS = 10;
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomFigureNumber = random.nextInt(FigureShape.values().length);
+
+        switch (FigureShape.values()[randomFigureNumber]) {
+            case SQUARE:
+                return randomSquare();
+            case RECTANGLE:
+                return randomRectangle();
+            case RIGHT_TRIANGLE:
+                return randomRightTriangle();
+            case ISOSCELES_TRAPEZOID:
+                return randomIsoscelesTrapezoid();
+            default:
+                return randomCircle();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, DEFAULT_CIRCLE_RADIUS);
+    }
+
+    private Square randomSquare() {
+        double side = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        return new Square(colorSupplier.getRandomColor(), side);
+    }
+
+    private Circle randomCircle() {
+        double radius = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        return new Circle(colorSupplier.getRandomColor(), radius);
+    }
+
+    private Rectangle randomRectangle() {
+        double length = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        double width = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        return new Rectangle(colorSupplier.getRandomColor(), length, width);
+    }
+
+    private RightTriangle randomRightTriangle() {
+        double firstLeg = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        double secondLeg = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        return new RightTriangle(colorSupplier.getRandomColor(), firstLeg, secondLeg);
+    }
+
+    private IsoscelesTrapezoid randomIsoscelesTrapezoid() {
+        double longerBase;
+        double shorterBase;
+        double height = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        double baseSide1 = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+        double baseSide2 = Math.round((MAX_VALUE * random.nextDouble() + MIN_VALUE)
+                * DECIMAL_FORMAT) / DECIMAL_FORMAT;
+
+        if (baseSide1 > baseSide2) {
+            longerBase = baseSide1;
+            shorterBase = baseSide2;
+        } else {
+            longerBase = baseSide2;
+            shorterBase = baseSide1;
+        }
```

Comment:
I don't see any reasonable difference except for fields naming in the model to make it that complicated, we can get rid of all this just by changing the name to something more neutral, like `BaseA` and `BaseB` for e.g.